File: src/application/use-cases/ClassifyPatient.ts
import { VitalSigns, TriageResult } from '@/domain/entities/Patient';
import { ITriageService } from '@/domain/ports/ITriageService';

export class ClassifyPatientUseCase {
  constructor(private readonly triageService: ITriageService) {}

  async execute(
    symptoms: string[],
    vitalSigns: VitalSigns,
    painScale: number,
    medicalHistory: string[]
  ): Promise<TriageResult> {
    return this.triageService.classify(symptoms, vitalSigns, painScale, medicalHistory);
  }
}

File: backend/app/application/use_cases/classify_patient.py
"""
Use case that receives patient symptoms and vital signs to return a triage classification result.
"""

from app.domain.services.triage_service import TriageService
from app.domain.entities.patient import VitalSigns, TriageResult
from typing import List

class ClassifyPatientUseCase:
    def __init__(self, triage_service: TriageService):
        self._triage_service = triage_service

    def execute(
        self,
        symptoms: List[str],
        vital_signs: VitalSigns,
        pain_scale: int,
        medical_history: List[str]
    ) -> TriageResult:
        return self._triage_service.classify_patient(
            symptoms=symptoms,
            vital_signs=vital_signs,
            pain_scale=pain_scale,
            medical_history=medical_history
        )